security/WFLY-15199_Remove_Legacy_Security_Subsystem.adoc

```diff
@@ -0,0 +1,287 @@
+= [WFLY-15199] Remove legacy the security subystsem from the feature packs and convert to a skeleton,
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+During the development of WildFly 11 a new security solution using WildFly Elytron was integrated
+into WildFly in parallel with the legacy security solution based on PicketBox.  This enhancement 
+is to complete the activities to de-activate the legacy security subsystem and remove it from the
+default configurations and Galleon feature packs.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-15199[WFLY-15199 - Remove legacy security configuration from all feature packs.]
+* https://issues.redhat.com/browse/WFLY-15301[WFLY-15301 - Convert the legacy security subsystem to a model only subsystem.]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1094[EAP7-1094]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+The `security` subsystem will be converted to a skeleton subsystem, this will mean that it can
+be used in domain mode to manage older servers but it can not be used for runtime configuration in
+WildFly 25 or later.
+
+All feature packs will be updated to remove the legacy security subsystem and any references to
+it's capabilities.
+
+The following subsystems will require additional configuration to default to WildFly Elytron
```

Comment:
Are there any other subsystems to be adjusted?

Reply:
I have found roughly the following

- /subsystem=datasources/data-source, security-domain, not deprecated
- /subsystem=datasources/data-source=ExampleDS, security-domain, not deprecated
- /subsystem=datasources/xa-data-source, security-domain, not deprecated
- /subsystem=ejb3, default-security-domain, not deprecated
- /subsystem=iiop-openjdk, security-domain, not deprecated
- /subsystem=messaging-activemq/server, security-domain, deprecated
- /subsystem=resource-adapters/resource-adapter=a/connection-definitions, recovery-security-domain, security-application, security-domain, security-domain-and-application, not deprecated
- /subsystem=undertow, default-security-domain, not deprecated
- /deployment=a/subsystem=resource-adapters, similar to /subsystem=resource-adapters/resource-adapter=a/connection-definitions

Reply:
@OndrejKotek I see your list here, I am just going to try and check them out, if we have missed any that need doing I will move into JBEAP - or we could move the whole list into JBEAP and work through it there?

Reply:
As discussed on the meeting, let's move into JBEAP.

Reply:
https://issues.redhat.com/browse/JBEAP-24586 and https://issues.redhat.com/browse/WFLY-17659